pkg/cli/cli.go

```diff
@@ -175,6 +182,37 @@ func (c *cli) initialize() error {
 	return nil
 }
 
+// validate validates fields in a cli.
+func (c cli) validate() error {
+	// Validate project versions.
+	if err := validation.ValidateProjectVersion(c.defaultProjectVersion); err != nil {
+		return fmt.Errorf("failed to validate default project version %q: %v", c.defaultProjectVersion, err)
+	}
+	if err := validation.ValidateProjectVersion(c.projectVersion); err != nil {
+		return fmt.Errorf("failed to validate project version %q: %v", c.projectVersion, err)
+	}
+
+	// Validate plugin versions and name.
+	for _, versionedPlugins := range c.plugins {
+		for _, versionedPlugin := range versionedPlugins {
+			pluginName := versionedPlugin.Name()
```

Comment:
```suggestion
```

Reply:
I think we do not need the var.

Reply:
Both `versionedPlugin.Name()` and `versionedPlugin.Version()` are used twice. Three of those four usages are in error messages (where having to access it again will not impact performance as we already found an error). I don't really have a strong opinion about this, but I would lean slightly towards not declaring variables.

Reply:
Going to leave this as-is.